src/UmbracoExamine.PDF/PdfIndexPopulator.cs

```diff
@@ -32,12 +33,99 @@ public PdfIndexPopulator(IMediaService mediaService, IPdfIndexValueSetBuilder me
         /// <param name="parentId"></param>
         /// <param name="mediaService"></param>
         /// <param name="mediaValueSetBuilder"></param>
-        public PdfIndexPopulator(int? parentId, IMediaService mediaService, IPdfIndexValueSetBuilder mediaValueSetBuilder)
+        public PdfIndexPopulator(int? parentId, IMediaService mediaService, IPdfIndexValueSetBuilder mediaValueSetBuilder, IExamineManager examineManager)
         {
             _parentId = parentId;
             _mediaService = mediaService;
             _mediaValueSetBuilder = mediaValueSetBuilder;
+            _examineManager = examineManager;
             RegisterIndex("PDFIndex");
+
+            //Tie into various events one media saved/trashed/deleted/restored
+            Umbraco.Core.Services.Implement.MediaService.Saved += MediaService_Saved;
+            Umbraco.Core.Services.Implement.MediaService.Deleted += MediaService_Deleted;
+            Umbraco.Core.Services.Implement.MediaService.Trashed += MediaService_Trashed;
+            Umbraco.Core.Services.Implement.MediaService.Moved += MediaService_Moved;
+        }
+
+
+        /// <summary>
+        /// If pdfs were restored form recycle bin, re-index them
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="e"></param>
+        private void MediaService_Moved(IMediaService sender, Umbraco.Core.Events.MoveEventArgs<IMedia> e)
+        {
+            var restored = e.MoveInfoCollection
+                .Where(m => m.OriginalPath.StartsWith("-1")) //recyclebin
```

Comment:
Media's recycle bin is `-21`.

```suggestion
                .Where(m => m.OriginalPath.StartsWith(Constants.System.RecycleBinMediaString))
```